test/mlir/onnx/onnx_lowering.mlir

```diff
@@ -1753,3 +1753,314 @@ func @test_maxpool_pooling_operation(%arg0 : tensor<1x3x32x32xf32>) -> tensor<*x
+  // CHECK:        {{.*}} = mulf [[Ht1_LOAD]], [[Ri_LOAD]] : f32
+  // CHECK:        {{.*}} = load [[Ht1Ri_GEMM]][] : memref<f32>
+  // CHECK:        {{.*}} = addf {{.*}}, {{.*}} : f32
+  // CHECK:        store %63, [[Ht1Ri_GEMM]][] : memref<f32>
+
+  // CHECK:        [[Wo_LOAD:%.+]] = load %arg1[%c0, [[OUTPUT_HIDDEN_INDEX]], %arg6] : memref<1x12x2xf32>
+  // CHECK:        {{.*}} = mulf [[Xt_LOAD]], [[Wo_LOAD]] : f32
+  // CHECK:        {{.*}} = load [[XtWo_GEMM]][] : memref<f32>
+  // CHECK:        {{.*}} = addf {{.*}}, {{.*}} : f32
+  // CHECK:        store %67, [[XtWo_GEMM]][] : memref<f32>
+
+  // CHECK:        [[Ro_LOAD:%.+]] = load %arg2[%c0, [[OUTPUT_HIDDEN_INDEX]], %arg6] : memref<1x12x3xf32>
+  // CHECK:        {{.*}} = mulf [[Ht1_LOAD]], [[Ro_LOAD]] : f32
+  // CHECK:        {{.*}} = load [[Ht1Ro_GEMM]][] : memref<f32>
+  // CHECK:        {{.*}} = addf {{.*}}, {{.*}} : f32
+  // CHECK:        store %71, [[Ht1Ro_GEMM]][] : memref<f32>
+
+  // CHECK:        [[Wf_LOAD:%.+]] = load %arg1[%c0, [[FORGET_HIDDEN_INDEX]], %arg6] : memref<1x12x2xf32>
+  // CHECK:        {{.*}} = mulf [[Xt_LOAD]], [[Wf_LOAD]] : f32
+  // CHECK:        {{.*}} = load [[XtWf_GEMM]][] : memref<f32>
+  // CHECK:        {{.*}} = addf {{.*}}, {{.*}} : f32
+  // CHECK:        store %75, [[XtWf_GEMM]][] : memref<f32>
+
+  // CHECK:        [[Rf_LOAD:%.+]] = load %arg2[%c0, [[FORGET_HIDDEN_INDEX]], %arg6] : memref<1x12x3xf32>
+  // CHECK:        {{.*}} = mulf [[Ht1_LOAD]], [[Rf_LOAD]] : f32
+  // CHECK:        {{.*}} = load [[Ht1Rf_GEMM]][] : memref<f32>
+  // CHECK:        {{.*}} = addf {{.*}}, {{.*}} : f32
+  // CHECK:        store %79, [[Ht1Rf_GEMM]][] : memref<f32>
+
+  // CHECK:        [[Wc_LOAD:%.+]] = load %arg1[%c0, [[CELL_HIDDEN_INDEX]], %arg6] : memref<1x12x2xf32>
+  // CHECK:        {{.*}} = mulf [[Xt_LOAD]], [[Wc_LOAD]] : f32
+  // CHECK:        {{.*}} = load [[XtWc_GEMM]][] : memref<f32>
+  // CHECK:        {{.*}} = addf {{.*}}, {{.*}} : f32
+  // CHECK:        store %83, [[XtWc_GEMM]][] : memref<f32>
+
+  // CHECK:        [[Rc_LOAD:%.+]] = load %arg2[%c0, [[CELL_HIDDEN_INDEX]], %arg6] : memref<1x12x3xf32>
+  // CHECK:        {{.*}} = mulf [[Ht1_LOAD]], [[Rc_LOAD]] : f32
+  // CHECK:        {{.*}} = load [[Ht1Rc_GEMM]][] : memref<f32>
+  // CHECK:        {{.*}} = addf {{.*}}, {{.*}} : f32
+  // CHECK:        store %87, [[Ht1Rc_GEMM]][] : memref<f32>
+  // CHECK:      }
+
+  // CHECK:      [[XtWi_LOAD:%.+]] = load [[XtWi_GEMM]][] : memref<f32>
+  // CHECK:      [[Ht1Ri_LOAD:%.+]] = load [[Ht1Ri_GEMM]][] : memref<f32>
+  // CHECK:      [[It_OUTPUT:%.+]] = addf [[XtWi_LOAD]], [[Ht1Ri_LOAD]] : f32
+
+  // CHECK:      [[SIGMOID_INPUT:%.+]] = alloc() : memref<f32>
+  // CHECK:      store [[It_OUTPUT]], [[SIGMOID_INPUT]][] : memref<f32>
+  // CHECK:      krnl.define_loops 0
+  // CHECK:      krnl.optimize_loops  {
+  // CHECK:        krnl.return_loops
+  // CHECK:      } : () -> ()
+  // CHECK:      krnl.iterate() with () {
```

Comment:
We seem to have a degenerate loop here, as a consequence of calling the activation op with a degenerate memref. Do you think it makes sense to specialize the Op lowering logic for element-wise operations to not emit loops when MemRef is a scalar?

Reply:
Yes, it makes a lot of sense.